src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -158,6 +158,10 @@ var DOMPropertyOperations = {
         } else if (propertyInfo.hasBooleanValue ||
                    (propertyInfo.hasOverloadedBooleanValue && value === true)) {
           node.setAttribute(attributeName, '');
+        } else if (attributeName === 'value') {
+          if (node.value != value) {
+            node.value = '' + value;
+          }
```

Comment:
Ack! I crossed the streams!

This PR was supposed to be solely for fixing `defaultValue`, but it looks I got my branches mixed up and pushed the more general fix for both inputs. I actually just replied here:

https://github.com/facebook/react/issues/7253#issuecomment-235602919

Again, pushing to this branch was a total accident. A bunch of tests fail now. I'll work on getting those to pass in the mean time, but could definitely use some advice.